aspnetcore/security/authentication/azure-ad-b2c.md

```diff
@@ -4,150 +4,124 @@ author: camsoper
-    |-------------------------------|-------------------------------------------------------|
-    | **Domain Name**               | *&lt;the domain name of your B2C tenant&gt;*          |
-    | **Application ID**            | *&lt;paste the Application ID from the clipboard&gt;* |
-    | **Callback Path**             | *&lt;use the default value&gt;*                       |
-    | **Sign-up or sign-in policy** | `B2C_1_SiUpIn`                                        |
-    | **Reset password policy**     | `B2C_1_SSPR`                                          |
-    | **Edit profile policy**       | *&lt;leave blank&gt;*                                 |
-    
-    Select the **Copy** link next to **Reply URI** to copy the Reply URI to the clipboard. Select **OK** to close the **Change Authentication** dialog. Select **OK** to create the web app.
+    :::code language="razor" source="azure-ad-b2c/sample/Pages/Shared/_LoginPartial.cshtml":::    
 
-## Finish the B2C app registration
+    In the preceding code:
 
-Return to the browser window with the B2C app properties still open. Change the temporary **Reply URL** specified earlier to the value copied from Visual Studio. Select **Save** at the top of the window.
+    - The partial view checks if the user is authenticated.
+    - Renders a "Sign out" or "Sign in" link as appropriate.
+    - The link points to an action method on the `Account` controller in the `MicrosoftIdentity` area.
 
-> [!TIP]
-> If you didn't copy the Reply URL, use the HTTPS address from the Debug tab in the web project properties, and append the **CallbackPath** value from *appsettings.json*.
+1. In *Views/Shared/_Layout.cshtml*, add the highlighted line within the `<header>` element:
 
-## Configure policies
+    :::code language="razor" source="azure-ad-b2c/sample/Pages/Shared/_Layout.cshtml" range="11-32" highlight="10":::
+ 
+    This renders the *_LoginPartial.cshtml* partial view in every page request that uses this layout.
 
-Use the steps in the Azure AD B2C documentation to [create a sign-up or sign-in policy](/azure/active-directory-b2c/active-directory-b2c-reference-policies#user-flow-versions), and then [create a password reset policy](/azure/active-directory-b2c/active-directory-b2c-reference-policies#user-flow-versions). Use the example values provided in the documentation for **Identity providers**, **Sign-up attributes**, and **Application claims**. Using the **Run now** button to test the policies as described in the documentation is optional.
+1. In *Startup.cs*, make the following changes:
 
-> [!WARNING]
-> Ensure the policy names are exactly as described in the documentation, as those policies were used in the **Change Authentication** dialog in Visual Studio. The policy names can be verified in *appsettings.json*.
+    1. Add the following `using` directives:
+    
+        :::code language="csharp" source="azure-ad-b2c/sample/Startup.cs" range="11-13":::
```

Comment:
can we use snippets? Much cleaner. When code changes, no need to recalculate.

Reply:
Done